pkg/controller/controller_test.go

```diff
@@ -361,6 +362,42 @@ func TestController(t *testing.T) {
 				"testPlugin": testPlugin,
 			},
 		},
+		"plugin not processed if spec invalid according to schema": &testCase{
+			bundle: &smith_v1.Bundle{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name:      "bundle1",
+					Namespace: testNamespace,
+					UID:       "uid123",
+				},
+				Spec: smith_v1.BundleSpec{
+					Resources: []smith_v1.Resource{
+						{
+							Name: "p1",
+							Spec: smith_v1.ResourceSpec{
+								Plugin: &smith_v1.PluginSpec{
+									Name:       "testPlugin",
+									ObjectName: "m1",
+									Spec: map[string]interface{}{
+										"p1": nil,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			plugins: map[smith_v1.PluginName]func(*testing.T) plugin.Plugin{
+				"testPlugin": testPlugin,
+			},
+			test: func(t *testing.T, ctx context.Context, cntrlr *BundleController, tc *testCase) {
+				key, err := cache.MetaNamespaceKeyFunc(tc.bundle)
+				require.NoError(t, err)
+				retriable, err := cntrlr.processKey(key)
+				// Sadly, the actual error is not current propagated
```

Comment:
The error you are looking for is in the status of the bundle in a condition for the failed resource. See other tests.